lib/DBD/File.pm

```diff
@@ -109,7 +109,9 @@ sub connect
     # We do not (yet) care about conflicting attributes here
     # my $dbh = DBI->connect ("dbi:CSV:f_dir=test", undef, undef, { f_dir => "text" });
     # will test here that both test and text should exist
-    if (my $attr_hash = (DBI->parse_dsn ($dbname))[3]) {
+    if ($dbname) {
+	my @attrs = split ";", $dbname;
```

Comment:
I added the final line of commit message as comment and fixed the split call.
I have no idea whether a ':' can occur in any of the attribute values, in the name it's not likely that it will remain a valid attribute name.